### ISSUE TITLE
`__slots__` check doesn't properly handle descriptors stored as class variables

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3172]
```python
from typing import Any

class MyDescriptor:
    def __init__(self, *, slot: str):
        ...

    def __set__(self, instance: object, value: object) -> None:
        ...

    def __get__(self, instance: object, owner: Any) -> Any:
        ...

class ClassA:
    foo = MyDescriptor(slot='_foo_weak')
    __slots__ = ('_foo_weak')

    def __init__(self, foo: int) -> None:
        self.foo = foo # False positive: "foo" is not specified in __slots__
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.229, which I just published. It will also be included in the next release of pylance.

**Comment by jab** [comment]
This bug still appears to be occurring in pyright 1.1.229 for descriptors that are inherited:
```python
from typing import Any

class MyDescriptor:
    def __init__(self, *, slot: str):
        ...

    def __set__(self, instance: object, value: object) -> None:
        ...

    def __get__(self, instance: object, owner: Any) -> Any:
        ...

class ClassA:
    foo = MyDescriptor(slot='_foo_weak')
    __slots__ = ('_foo_weak',)


class ClassB(ClassA):
    __slots__ = ()
    def __init__(self, foo: int) -> None:
        self.foo = foo # False positive: "foo" is not specified in __slots__
```

**Comment by erictraut** [comment]
I'm not able to repro any error with the code snippet above. If you think there's another bug here, please open a new issue.

**Comment by jab** [comment]
Sorry about that! Please see #3183.
